Provide a reliable way to perform form-based authentication logout (#46429)
quarkusio/quarkus: docs/src/main/asciidoc/security-authentication-mechanisms.adoc

    if (identity.getIdentity().isAnonymous()) {
        throw new UnauthorizedException("Not authenticated");
    }
    FormAuthenticationMechanism.logout(identity.getIdentity()); <1>
    return Response.noContent().build();
}
----

> <1> For reliable logout, destroy persistent session cookie with this utility method.

Reviewer: ```suggestion
<1> Perform the logout by removing the session cookie.
```
